apps/veil/src/pages/tournament/ui/vote-dialog/index.tsx

```diff
@@ -0,0 +1,172 @@
+import { voteTournament } from '../../api/vote';
+import { useCurrentEpoch } from '../../api/use-current-epoch';
+import { useEpochGauge } from '../../api/use-epoch-gauge';
+import { MappedGauge } from '../../server/previous-epochs';
+import { VoteDialogSearchResults } from './search-results';
+import { VotingAssetSelector } from './asset-selector';
+
+interface VoteDialogProps {
+  defaultValue?: string;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const DEFAULT_REVEAL_VOTE = true;
+
+export const VoteDialogueSelector = observer(
+  ({ isOpen, onClose, defaultValue }: VoteDialogProps) => {
+    const [selectedDenom, setSelectedDenom] = useState<string | undefined>(defaultValue);
+    const [selectedAsset, setSelectedAsset] = useState<MappedGauge | undefined>();
+    const [revealVote, setRevealVote] = useState(DEFAULT_REVEAL_VOTE);
+
+    const [isSearchOpen, setIsSearchOpen] = useState(false);
+    const [searchQuery, setSearchQuery] = useState('');
+
+    const { data: subaccounts } = useSubaccounts();
+
+    const { subaccount } = connectionStore;
+    const valueAddress = subaccounts?.find(
+      account => getAddressIndex(account).account === subaccount,
+    );
+
+    // Temporarily hardcode the same account address as the reward recipient.
+    const rewardsRecipient = valueAddress?.addressView.value?.address;
+
+    // Fetch user's spendable voting notes for this epoch
+    const { notes } = useLQTNotes(subaccount);
+    const { epoch } = useCurrentEpoch();
+    const { data: assets, isLoading } = useEpochGauge(epoch);
+
+    const handleVoteSubmit = async () => {
+      if (selectedDenom) {
+        if (!epoch) {
+          throw new Error('Missing epoch index');
+        }
+
+        const stakedNotes: SpendableNoteRecord[] = notes
+          ? Array.from(notes.values())
+              .map(res => res.noteRecord)
+              .filter((record): record is SpendableNoteRecord => !!record)
+          : [];
+
+        // Craft LQT TPR and submit vote
+        await voteTournament({
+          stakedNotes: stakedNotes,
+          incentivized: selectedDenom,
+          epochIndex: epoch,
+          rewardsRecipient,
+        });
+
+        handleClose();
+      }
+    };
+
+    const handleClose = () => {
+      setSearchQuery('');
+      setSelectedDenom(undefined);
+      setSelectedAsset(undefined);
+      setIsSearchOpen(false);
+      setRevealVote(DEFAULT_REVEAL_VOTE);
+      onClose();
+    };
+
+    const onSearchSelect = (asset: MappedGauge) => {
+      setSearchQuery('');
+      setIsSearchOpen(false);
+      setSelectedDenom(asset.asset.base);
+      setSelectedAsset(asset);
+    };
+
+    return (
+      <Dialog isOpen={isOpen} onClose={handleClose}>
+        <Dialog.Content
+          title={`Vote in epoch #${epoch}`}
+          headerChildren={
+            <>
+              {/* Focus catcher. If this button wouldn't exist, the focus would go to the first input, which is undesirable */}
```

Comment:
Did you try to wrap this in a div with tabIndex="-1"?

Reply:
Tabindex=-1 has a negative side-effect of the input not being focusable afterwards. This catcher might look weird in the code, but it does what it needs to do